src/gogma/overlap.jl

```diff
@@ -105,7 +105,7 @@ end
 function force!(f::AbstractVector, x::AbstractIsotropicGaussian, y::AbstractIsotropicGMM, pσ=nothing, pϕ=nothing; kwargs...)
     if isnothing(pσ) && isnothing(pϕ)
         pσ = x.σ^2 .+ [gy.σ^2 for gy in y.gaussians]
-        pϕ = [ x.ϕ * gy.ϕ for gy in y.gaussians]
+        pϕ = x.ϕ * [gy.ϕ for gy in y.gaussians]
```

Comment:
The original was more efficient. The "new" form allocates two vectors: one temporary vector for `[gy.ϕ for gy in y.gaussians]` and the second for the output vector when multiplying by `x.ϕ`. In contrast, the first form creates just one vector. Scalars use CPU registers and hence are cheap, but arrays involve the memory allocator/garbage collector and are thus much more expensive. Even in C/C++, `malloc/free` are expensive, roughly commensurate with Julia's garbage collector.

Reply:
Yes, probably better to put it inside the comprehension.